app/src/main/java/com/paulmandal/atak/forwarder/plugin/ui/viewmodels/ChannelStatusViewModel.java

```diff
@@ -66,13 +66,19 @@ protected void updateSettings(SharedPreferences sharedPreferences) {
 
     @Override
     protected void complexUpdate(SharedPreferences sharedPreferences, String key) {
-        String channelName = sharedPreferences.getString(PreferencesKeys.KEY_CHANNEL_NAME, PreferencesDefaults.DEFAULT_CHANNEL_NAME);
-        int channelMode = Integer.parseInt(sharedPreferences.getString(PreferencesKeys.KEY_CHANNEL_MODE, PreferencesDefaults.DEFAULT_CHANNEL_MODE));
-        byte[] psk = Base64.decode(sharedPreferences.getString(PreferencesKeys.KEY_CHANNEL_PSK, PreferencesDefaults.DEFAULT_CHANNEL_PSK), Base64.DEFAULT);
-        ChannelProtos.ChannelSettings.ModemConfig modemConfig = ChannelProtos.ChannelSettings.ModemConfig.forNumber(channelMode);
-
-        mChannelName.postValue(channelName);
-        mModemConfig.postValue(modemConfig);
-        mPskHash.postValue(mHashHelper.hashFromBytes(psk));
+        // This is in complexUpdate because simpleUpdate is called before the MutableLiveData are available
+        List<ChannelConfig> channelConfigs = new Gson().fromJson(sharedPreferences.getString(PreferencesKeys.KEY_CHANNEL_DATA, PreferencesDefaults.DEFAULT_CHANNEL_DATA), new TypeToken<ArrayList<ChannelConfig>>() {}.getType());
+
+        if (channelConfigs == null) {
+            return;
+        }
+
```

Comment:
This should default to the first channel in case there is no channel marked `default`